base_view_inheritance_extension/tests/test_base_view_inheritance_extension.py

```diff
@@ -0,0 +1,239 @@
+                    {"default_company_id": company_id}
+                </attribute>
+            </field>
+            """
+        )
+        res = self.env["ir.ui.view"].apply_inheritance_specs(source, specs)
+        self.assertEqual(
+            res.xpath('//field[@name="account_move_id"]')[0].attrib["context"],
+            "{'default_journal_id': journal_id, 'default_company_id': company_id}",
+        )
+
+    def test_update_context_complex(self):
+        source = etree.fromstring(
+            """
+            <form>
+                <field
+                    name="invoice_line_ids"
+                    context="{
+                        'default_type': context.get('default_type'),
+                        'journal_id': journal_id,
+                        'default_partner_id': commercial_partner_id,
+                        'default_currency_id': (
+                            currency_id != company_currency_id and currency_id or False
+                        ),
+                        'default_name': 'The company name',
+                    }"
+                />
+            </form>
+            """
+        )
+        specs = etree.fromstring(
+            """
+            <field name="invoice_line_ids" position="attributes">
+                <attribute name="context" operation="update">
+                    {
+                        "default_product_id": product_id,
+                        "default_cost_center_id": (
+                            context.get("handle_mrp_cost") and cost_center_id or False
+                        ),
+                    }
+                </attribute>
+            </field>
+            """
+        )
+        res = self.env["ir.ui.view"].apply_inheritance_specs(source, specs)
+        expected_items = [
+            "'default_type': context.get('default_type')",
+            "'journal_id': journal_id",
+            "'default_partner_id': commercial_partner_id",
+            (
+                "'default_currency_id': "
+                "currency_id != company_currency_id and currency_id or False"
+            ),
+            "'default_name': 'The company name'",
+            "'default_product_id': product_id",
+            (
+                "'default_cost_center_id': "
+                "context.get('handle_mrp_cost') and cost_center_id or False"
+            ),
+        ]
+        self.assertEqual(
+            res.xpath('//field[@name="invoice_line_ids"]')[0].attrib["context"],
+            "{%s}" % ", ".join(expected_items),  # noqa: UP031
+        )
+
+    def test_update_invisible_domain_add(self):
+        """Test that we can add a domain to an existing invisible attribute"""
+        source = etree.fromstring(
+            """
+            <form>
+                <field
+                    name="ref"
+                    invisible="[('state', '=', 'draft')]"
+                    attrs="{'invisible': [('state', '=', 'draft')]}"
```

Comment:
> The XML attributes attrs and states are no longer used, and should be replaced by their equivalent Python expression under invisible, required and/or readonly attributes. Example: attrs="{'invisible': [('name', 

so this case is outdated

Reply:
removed this test-case

Reply:
the content of the test case might be outdated, but we still need to test `operation="update"` no?

Reply:
allo?

Reply:
> the content of the test case might be outdated, but we still need to test `operation="update"` no?

I have change test-case use `operation="update"` for `context` field which is use `dict` like `attrs` to make it suitable